packages/order-utils/src/signature_utils.ts

```diff
@@ -225,57 +225,96 @@ export async function ecSignOrderHashAsync(
     // v + r + s OR r + s + v, and different clients (even different versions of the same client)
     // return the signature params in different orders. In order to support all client implementations,
     // we parse the signature in both ways, and evaluate if either one is a valid signature.
+    // r + s + v is the most prevalent format from eth_sign, so we attempt this first.
     // tslint:disable-next-line:custom-no-magic-numbers
     const validVParamValues = [27, 28];
+    const ecSignatureRSV = parseSignatureHexAsRSV(signature);
+    if (_.includes(validVParamValues, ecSignatureRSV.v)) {
+        const isValidRSVSignature = isValidECSignature(prefixedMsgHashHex, ecSignatureRSV, normalizedSignerAddress);
+        if (isValidRSVSignature) {
+            const convertedSignatureHex = convertECSignatureToSignatureHex(ecSignatureRSV, signerType);
+            return convertedSignatureHex;
+        }
+    }
     const ecSignatureVRS = parseSignatureHexAsVRS(signature);
     if (_.includes(validVParamValues, ecSignatureVRS.v)) {
         const isValidVRSSignature = isValidECSignature(prefixedMsgHashHex, ecSignatureVRS, normalizedSignerAddress);
         if (isValidVRSSignature) {
-            return ecSignatureVRS;
+            const convertedSignatureHex = convertECSignatureToSignatureHex(ecSignatureVRS, signerType);
+            return convertedSignatureHex;
         }
     }
 
-    const ecSignatureRSV = parseSignatureHexAsRSV(signature);
-    if (_.includes(validVParamValues, ecSignatureRSV.v)) {
-        const isValidRSVSignature = isValidECSignature(prefixedMsgHashHex, ecSignatureRSV, normalizedSignerAddress);
-        if (isValidRSVSignature) {
-            return ecSignatureRSV;
+    throw new Error(OrderError.InvalidSignature);
+}
+/**
+ * Combines ECSignature with V,R,S and the relevant signature type for use in 0x protocol
+ * @param ecSignature The ECSignature of the signed data
+ * @param signerType The SignerType of the signed data
+ * @return Hex encoded string of signature (v,r,s) with Signature Type
+ */
+export function convertECSignatureToSignatureHex(ecSignature: ECSignature, signerType: SignerType): string {
+    const signatureBuffer = Buffer.concat([
+        ethUtil.toBuffer(ecSignature.v),
+        ethUtil.toBuffer(ecSignature.r),
+        ethUtil.toBuffer(ecSignature.s),
+    ]);
+    const signatureHex = `0x${signatureBuffer.toString('hex')}`;
+    switch (signerType) {
+        case SignerType.Metamask:
+        case SignerType.Ledger:
+        case SignerType.Default: {
+            const signatureType = SignatureType.EthSign;
+            const signatureWithType = convertToSignatureWithType(signatureHex, signatureType);
+            return signatureWithType;
+        }
+        case SignerType.Trezor: {
+            const signatureType = SignatureType.Trezor;
+            const signatureWithType = convertToSignatureWithType(signatureHex, signatureType);
+            return signatureWithType;
```

Comment:
Rather then return, lets move the body of `convertToSignatureWithType` below the switch statement.